aclavijo-task/Backend/src/main/java/com/workshop/awscognitoidp/controllers/TrainerController.java

```diff
@@ -0,0 +1,154 @@
+package com.workshop.awscognitoidp.controllers;
+
+import com.workshop.awscognitoidp.models.Student;
+import com.workshop.awscognitoidp.models.Subject;
+import com.workshop.awscognitoidp.models.Trainer;
+import com.workshop.awscognitoidp.services.JWTDecoder;
+import com.workshop.awscognitoidp.services.JwtValidator;
+import com.workshop.awscognitoidp.services.StudentService;
+import com.workshop.awscognitoidp.services.SubjectService;
+import com.workshop.awscognitoidp.services.TrainerService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+
+@RestController
+@RequestMapping(path = "/api/trainers")
+public class TrainerController {
+    @Autowired
+    private JwtValidator jwtValidator;
+    @Autowired
+    private JWTDecoder JWTDecoder;
+    @Autowired
+    private SubjectService subjectService;
+    @Autowired
+    private StudentService studentService;
+    @Autowired
+    private TrainerService trainerService;
+
+    
+    @GetMapping("/students")
+    public List<Student> getStudents(@RequestHeader("Authorization") String token) {
+        if(jwtValidator.validateJwtToken(token.substring(7))){
+            
+            String role  = JWTDecoder.getRole(token.substring(7));
+            if (role.equals("Trainer") || role.equals("Admin")) {
+                return studentService.findAll();
+            } else {
+                throw new AccessDeniedException("Insufficient permissions");
+            }
+        }
+        else {
+            throw new AccessDeniedException("Invalid token");
```

Comment:
If you have another user to that you need to allow access, so, in that case, this logic needs to be modified and we would break the Open/Closed principle, so for that reason, this control should be in the Global Security Config.
Please refer to this [link ](https://kevcodez.de/posts/2020-03-26-secure-spring-boot-app-with-oauth2-aws-cognito/)to review an example